apps/frontend/src/libs/components/input/input.tsx

```diff
@@ -30,32 +31,60 @@ const Input = <T extends FieldValues>({
 	placeholder = "",
 	type = "text",
 }: Properties<T>): React.JSX.Element => {
+	const [typeState, setTypeState] = useState<Properties<never>["type"]>(type);
 	const { field } = useFormController({ control, name });
 
 	const error = errors[name]?.message;
 	const hasError = Boolean(error);
+	const isPasswordButtonShown = type === "password" && field.value;
+
+	const onPassAppearanceToggle = useCallback((): void => {
+		const state: "password" | "text" =
+			typeState === "password" ? "text" : "password";
+
+		setTypeState(state);
+	}, [typeState, setTypeState]);
+	const toggleButtonIcon: "eye" | "eyeOff" =
+		typeState === "text" ? "eyeOff" : "eye";
 
 	return (
 		<label className={styles["label"]}>
 			<span className={styles["label-caption"]}>{label}</span>
-			<input
-				autoComplete={autocomplete}
-				className={combineClassNames(
-					styles["input"],
-					hasError && styles["input-error"],
+			<span className={styles["input-container"]}>
+				<input
+					autoComplete={autocomplete}
+					className={combineClassNames(
+						styles["input"],
+						hasError && styles["input-error"],
+					)}
+					name={field.name}
+					onChange={field.onChange}
+					placeholder={placeholder}
+					type={typeState}
+					value={field.value}
+				/>
+				{hasError && (
+					<span className={styles["error"]}>
+						<Icon height={24} name="error" width={24} />
+						{error as string}
+					</span>
+				)}
+				{isPasswordButtonShown && field.value && (
```

Comment:
already have this check before
```suggestion
				{isPasswordButtonShown && (
```